nixos/modules/services/networking/prosody.nix

```diff
@@ -422,6 +422,13 @@ in
         description = "List of administrators of the current host";
       };
 
+      authentication = mkOption {
+        type = types.str;
+        default = "internal_hashed";
```

Comment:
If we change the default now, how would this affecting existing installations?

Reply:
As it says in the commit message:
`On existing
installations the next time a users user accounts will automatically
be upgraded from plain to hashed one-by-one as they log in.`

Reply:
The only installations it will create problems on are those that had set the `authentication` option before using `extraConfig`.